File: app/src/main/java/ru/ra66it/updaterforspotify/notification/NotificationDownloadService.java
package ru.ra66it.updaterforspotify.notification;

import android.app.IntentService;
import android.app.NotificationManager;
import android.content.Context;
import android.content.Intent;
import android.support.annotation.Nullable;

import ru.ra66it.updaterforspotify.utils.UtilsDownloadSpotify;


/**
 * Created by 2Rabbit on 30.09.2017.
 */

public class NotificationDownloadService extends IntentService {
    private static final String TAG = "NotificationDownloadService";
    public static final String ACTION_DOWNLOAD = "ACTION_DOWNLOAD";

    public NotificationDownloadService() {
        super(TAG);
    }

    @Override
    protected void onHandleIntent(@Nullable Intent intent) {
        final String action = intent.getAction();
        final String link = intent.getStringExtra(PollService.LATEST_LINK);
        final String name = intent.getStringExtra(PollService.LATEST_VERSION_NAME);
        int  id = intent.getIntExtra(PollService.NOTIFICATION_ID, 0);
        if (ACTION_DOWNLOAD.equals(action)) {
            //Hide Notification
            NotificationManager manager = (NotificationManager)
                    this.getSystemService(Context.NOTIFICATION_SERVICE);
            manager.cancel(id);
            UtilsDownloadSpotify.downloadSpotify(link, name);
        }
    }
}

File: app/src/main/java/ru/ra66it/updaterforspotify/presentation/service/NotificationDownloadService.kt
package ru.ra66it.updaterforspotify.presentation.service

import android.app.IntentService
import android.app.NotificationManager
import android.content.Context
import android.content.Intent
import ru.ra66it.updaterforspotify.actionDownload
import ru.ra66it.updaterforspotify.latestLinkKey
import ru.ra66it.updaterforspotify.latestVersionKey
import ru.ra66it.updaterforspotify.notificationIdKey
import ru.ra66it.updaterforspotify.presentation.utils.UtilsDownloadSpotify


/**
 * Created by 2Rabbit on 30.09.2017.
 */

class NotificationDownloadService : IntentService(NotificationDownloadService::class.java.simpleName) {

    override fun onHandleIntent(intent: Intent) {
        val action = intent.action
        val link = intent.getStringExtra(latestLinkKey)
        val id = intent.getIntExtra(notificationIdKey, 0)
        val version = intent.getStringExtra(latestVersionKey)
        if (actionDownload == action) {
            //Hide Notification
            val manager = this.getSystemService(Context.NOTIFICATION_SERVICE) as NotificationManager
            manager.cancel(id)
            UtilsDownloadSpotify.downloadSpotify(link, version)
        }
    }
}
